### PR TITLE
Improve description of Management Plane and Compass

### PR DESCRIPTION
**Pull request by PK85**
**Description**

Changes proposed in this pull request:

- diagram update - ui/serviceMesh and runtimeProvider added into ManagementPlane 
- the appearance of the diagram has been improved
- the terminology.md has been update updated. The Management Plane has better boundaries now.

**Related issue(s)**

### CODE REVIEW

File: docs/terminology.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+### MP Compass
+Abbr.: *Compass*\
+\
+Includes *Connector*, *Gateway*, *Director* and *Healtchecker* components.
+
+The multi-tenant system which allows to:
+- configure Applications
+- configure Runtimes
+- group Applications
+- assign Applications or Runtimes to the group
```
Comment (crabtree) [inline docs/terminology.md]: Aren't this point and the previous one the same? I would remove the third point.

File: docs/architecture/api-credentials-flows.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# API Credentials Flows
+
+## Overview
+
+This document describes how [Application](/docs/terminology.md#Application) provides and updates credentials for [API Definition](/docs/terminology.md#Application-API-Definiton) per [Runtime](/docs/terminology.md#Runtime).
+
+## Details
+This diagram illustrates the API credentials flow in details. The Application provides [Webhook API](/docs/terminology.md#Application-Webhook-API) where [Management Plane](/docs/terminology.md#Management-Plane) requests for providing new credentials for given API Definition.
+
+>**NOTE:** There is an option that Application do not support Webhook API. That means Application needs to monitor registered API Definitions and set API credentials when new Runtime assigned. The [Administrator](/docs/terminology.md#Administrator) can exchange credentials for registered APIs at any time too.
```
Comment (tgorgol) [inline docs/architecture/api-credentials-flows.md]:
```suggestion
>**NOTE:** There is an option that Application does not support Webhook API. That means Application needs to monitor registered API Definitions and set API credentials when new Runtime assigned. The [Administrator](/docs/terminology.md#Administrator) can exchange credentials for registered APIs at any time too.
```

File: docs/architecture/api-credentials-flows.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+![Application Webhook](./assets/api-credentials-flow.svg)
+
+Assume we have Application with one API Definiton which is already registered into Management Plane. No Runtimes assigned yet.
```
Comment (tgorgol) [inline docs/architecture/api-credentials-flows.md]:
```suggestion
Assume we have Application with one API Definition which is already registered into Management Plane. No Runtimes assigned yet.
```

File: docs/terminology.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
+# Terminology
+
+## Overview
+This document contains all terminology used across compass documentation.
+
+## Management Plane 
+Abbr.: *MP*\
+\
+Includes *Compass*, *Runtime Provisioners*, *ServiceMesh* and *UI* components.
```
Comment (tgorgol) [inline docs/terminology.md]:
```suggestion
Includes *Compass*, *Runtime Provisioners*, *Service Mesh* and *UI* components.
```

File: docs/terminology.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+The multi-tenant system which allows to:
+- create Applications
+- create Runtimes
+- manage Applications and Runtimes
+
+### MP Compass
+Abbr.: *Compass*\
+\
+Includes *Connector*, *Gateway*, *Director* and *Healtchecker* components.
```
Comment (tgorgol) [inline docs/terminology.md]: Maybe it should be mentioned that for now HealthCheck will be a part of Director?

Comment (PK85) [reply docs/terminology.md]: Done, I've remved healthchecker component from terminology and moved that information directly into Director.

File: docs/terminology.md [hunk 6]
```diff
@@ -6,36 +6,57 @@ This document contains all terminology used across compass documentation.
...
 Abbr.: *Connector*\
 \
-Component that establish trust among Applications, Management Plane and Runtimes. In first iteration we support only client certificates.
+Connector component establishes trust among Applications, Management Plane and Runtimes. In first iteration we support only client certificates.
 
 ### MP Gateway
 Abbr.: *Gateway*\
 \
-Component that.. TBD
+Gateway component serves as the main API Gateway that extracts token from incoming requests and proxies the requests to the Director component.
```
Comment (tgorgol) [inline docs/terminology.md]:
```suggestion
Gateway component serves as the main API Gateway that extracts Tenant from incoming requests and proxies the requests to the Director component.
```

**Pull request merged**